activity/activity_IngestDigestToEndpoint.py

```
@@ -0,0 +1,377 @@
+        if not digest_json:
+            self.logger.info(
+                "Did not get existing digest json from the endpoint for digest_id %s" %
+                str(digest_id))
+        self.digest_content = sync_json(self.digest_content, digest_json)
+        # set the stage attribute if missing
+        set_stage(self.digest_content)
+        self.logger.info("Digest stage value %s" % str(self.digest_content.get("stage")))
+
+        self.ingest_status = self.put_digest_to_endpoint(
+            digest_id, self.digest_content, self.settings)
+
+        self.emit_end_message(article_id, version, run)
+
+        return self.ACTIVITY_SUCCESS
+
+    def session_data(self, data):
+        "get session data and return basic values"
+        run = None
+        session = None
+        version = None
+        article_id = None
+        success = None
+        try:
+            run = data["run"]
+            session = get_session(self.settings, data, run)
+            version = session.get_value("version")
+            article_id = session.get_value("article_id")
+            success = True
+        except (TypeError, KeyError) as exception:
+            self.logger.exception("Exception when getting the session for Starting ingest digest " +
+                                  " to endpoint. Details: %s" % str(exception))
+            success = False
+        return success, run, session, article_id, version
+
+    def emit_message(self, article_id, version, run, status, message):
+        "emit message to the queue"
+        success = None
+        try:
+            self.emit_monitor_event(self.settings, article_id, version, run,
+                                    self.pretty_name, status, message)
+            success = True
+        except Exception as exception:
+            self.logger.exception("Exception emitting %s message. Details: %s" %
+                                  (str(status), str(exception)))
+            success = False
```

also here, can throw rather than return `False`